### PR TITLE
Replace LoggingFactory with StrictLogging pt. 2

### PR DESCRIPTION
**Pull request by zen-dog**
Summary:
In a few cases, we still used `LoggingFactory` instead of `StrictLogging`. Additionally got rid of a couple of compiler warnings.

### CODE REVIEW

File: mesos-simulation/src/main/scala/mesosphere/mesos/simulation/DriverActor.scala [hunk 1]
```diff
@@ -2,14 +2,14 @@ package mesosphere.mesos.simulation
 
 import java.util.UUID
 
-import akka.actor.{ Actor, ActorRef, Cancellable, Props }
+import akka.actor.{Actor, ActorRef, Cancellable, Props}
```
Comment (ichernetsky) [inline mesos-simulation/src/main/scala/mesosphere/mesos/simulation/DriverActor.scala]: Recompiling is warranted :)

File: src/main/scala/mesosphere/marathon/api/WebJarServlet.scala [hunk 2]
```diff
@@ -44,21 +42,19 @@ class WebJarServlet extends HttpServlet {
...
-         |path: ${req.getPathTranslated}
-         |uri: ${req.getRequestURI}
-         |jar: $jar
-         |resource: $resource
-         |file: $file
-         |mime: $mime
-         |resourceURI: $resourceURI
-       """.stripMargin)
-    }
+    logger.debug(
```
Comment (ichernetsky) [inline src/main/scala/mesosphere/marathon/api/WebJarServlet.scala]: Why `if` was removed? Without it there is string interpolation on each request.

File: src/main/scala/mesosphere/marathon/core/deployment/impl/DeploymentPlanReverter.scala [hunk 3]
```diff
@@ -66,11 +65,10 @@ private[deployment] object DeploymentPlanReverter {
       val addedDependencies = newGroup.dependencies -- oldGroup.dependencies
 
       if (removedDependencies.nonEmpty || addedDependencies.nonEmpty) {
-        if (log.isDebugEnabled)
-          log.debug(
-            s"revert dependency changes in group ${oldGroup.id}, " +
-              s"readding removed {${removedDependencies.mkString(", ")}}, " +
-              s"removing added {${addedDependencies.mkString(", ")}}")
+        logger.debug(
```
Comment (ichernetsky) [inline src/main/scala/mesosphere/marathon/core/deployment/impl/DeploymentPlanReverter.scala]: A similar case here too.

File: src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceUpdateActor.scala [hunk 4]
```diff
@@ -105,11 +103,9 @@ private[impl] class InstanceUpdateActor(
         operationsByInstanceId += op.instanceId -> newQueue
 
       val activeCount = metrics.numberOfActiveOps.decrement()
-      if (log.isDebugEnabled) {
-        val queuedCount = metrics.numberOfQueuedOps.value()
-        log.debug(s"Finished processing ${op.op} for app [${op.appId}] and ${op.instanceId} "
-          + s"$activeCount active, $queuedCount queued.")
-      }
+      val queuedCount = metrics.numberOfQueuedOps.value()
+      logger.debug(s"Finished processing ${op.op} for app [${op.appId}] and ${op.instanceId} "
```
Comment (ichernetsky) [inline src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceUpdateActor.scala]: The same.

File: src/main/scala/mesosphere/mesos/Constraints.scala [hunk 5]
```diff
@@ -235,17 +235,15 @@ object Constraints {
...
-        val attrs = instance.agentInfo.attributes.map(a => s"${a.getName}=${getValueString(a)}").mkString(", ")
-        s"${instance.instanceId} host:${instance.agentInfo.host} attrs:$attrs"
-      }.mkString("Selected Tasks to kill:\n", "\n", "\n")
-      val distDesc = distributions.map { d =>
-        val (before, after) = (d.distributionDifference(), d.distributionDifference(toKillInstances))
-        s"${d.constraint.getField} changed from: $before to $after"
-      }.mkString("Selected Constraint diff changed:\n", "\n", "\n")
-      log.info(s"$instanceDesc$distDesc")
-    }
+    val instanceDesc = toKillInstances.values.map { instance =>
```
Comment (ichernetsky) [inline src/main/scala/mesosphere/mesos/Constraints.scala]: ditto

File: src/main/scala/mesosphere/marathon/core/deployment/impl/DeploymentPlanReverter.scala [hunk 6]
```diff
@@ -91,13 +89,12 @@ private[deployment] object DeploymentPlanReverter {
...
         case Some(unchanged) if isGroupUnchanged(unchanged) =>
-          log.debug("remove unchanged group {}", unchanged.id)
+          logger.debug("remove unchanged group {}", unchanged.id)
           result.removeGroup(unchanged.id, version)
         case _ if newGroup.dependencies.nonEmpty =>
           // group dependencies have changed
-          if (log.isDebugEnabled)
-            log.debug(s"group ${newGroup.id} has changed. " +
-              s"Removed added dependencies ${newGroup.dependencies.mkString(", ")}")
+          logger.debug(s"group ${newGroup.id} has changed. " +
```
Comment (ichernetsky) [inline src/main/scala/mesosphere/marathon/core/deployment/impl/DeploymentPlanReverter.scala]: ditto

File: src/main/scala/mesosphere/marathon/core/launcher/impl/TaskLauncherImpl.scala [hunk 7]
```diff
@@ -25,9 +24,8 @@ private[launcher] class TaskLauncherImpl(
       //The filter duration is set to 0, so we get the same offer in the next allocator cycle.
       val noFilter = Protos.Filters.newBuilder().setRefuseSeconds(0).build()
       val operations = taskOps.flatMap(_.offerOperations)
-      if (log.isDebugEnabled) {
-        log.debug(s"Operations on $offerID:\n${operations.mkString("\n")}")
-      }
+      logger.debug(s"Operations on $offerID:\n${operations.mkString("\n")}")
```
Comment (ichernetsky) [inline src/main/scala/mesosphere/marathon/core/launcher/impl/TaskLauncherImpl.scala]: ditto

File: src/main/scala/mesosphere/marathon/core/launcher/impl/TaskLauncherImpl.scala [hunk 8]
```diff
@@ -58,13 +56,12 @@ private[launcher] class TaskLauncherImpl(
     marathonSchedulerDriverHolder.driver match {
       case Some(driver) =>
         val status = block(driver)
-        if (log.isDebugEnabled) {
-          log.debug(s"$description returned status = $status")
-        }
+        logger.debug(s"$description returned status = $status")
```
Comment (ichernetsky) [inline src/main/scala/mesosphere/marathon/core/launcher/impl/TaskLauncherImpl.scala]: ditto

File: mesos-simulation/src/main/scala/mesosphere/mesos/simulation/DriverActor.scala (hunk 1 above)
Comment (zen-dog) [reply mesos-simulation/src/main/scala/mesosphere/mesos/simulation/DriverActor.scala]: Do you know which part of linter/formatter does that? I remember trying to disable this once but I failed :(

**Pull request merged**